diff --git a/includes/bootstrap.inc b/includes/bootstrap.inc
index e42f542..203d4d6 100644
--- a/includes/bootstrap.inc
+++ b/includes/bootstrap.inc
@@ -2490,37 +2490,46 @@ function drupal_bootstrap($phase = NULL, $new_phase = TRUE) {
 
       switch ($current_phase) {
         case DRUPAL_BOOTSTRAP_CONFIGURATION:
+          if (function_exists('sb_stagemarker_message')) sb_stagemarker_message('BEFORE DRUPAL_BOOTSTRAP_CONFIGURATION');
           _drupal_bootstrap_configuration();
           break;
 
         case DRUPAL_BOOTSTRAP_PAGE_CACHE:
+          if (function_exists('sb_stagemarker_message')) sb_stagemarker_message('BEFORE DRUPAL_BOOTSTRAP_PAGE_CACHE');
           _drupal_bootstrap_page_cache();
           break;
 
         case DRUPAL_BOOTSTRAP_DATABASE:
+          if (function_exists('sb_stagemarker_message')) sb_stagemarker_message('BEFORE DRUPAL_BOOTSTRAP_DATABASE');
           _drupal_bootstrap_database();
           break;
 
         case DRUPAL_BOOTSTRAP_VARIABLES:
+          if (function_exists('sb_stagemarker_message')) sb_stagemarker_message('BEFORE DRUPAL_BOOTSTRAP_VARIABLES');
           _drupal_bootstrap_variables();
           break;
 
         case DRUPAL_BOOTSTRAP_SESSION:
+          if (function_exists('sb_stagemarker_message')) sb_stagemarker_message('BEFORE DRUPAL_BOOTSTRAP_SESSION');
           require_once DRUPAL_ROOT . '/' . variable_get('session_inc', 'includes/session.inc');
           drupal_session_initialize();
           break;
 
         case DRUPAL_BOOTSTRAP_PAGE_HEADER:
+          if (function_exists('sb_stagemarker_message')) sb_stagemarker_message('BEFORE DRUPAL_BOOTSTRAP_PAGE_HEADER');
           _drupal_bootstrap_page_header();
           break;
 
         case DRUPAL_BOOTSTRAP_LANGUAGE:
+          if (function_exists('sb_stagemarker_message')) sb_stagemarker_message('BEFORE DRUPAL_BOOTSTRAP_LANGUAGE');
           drupal_language_initialize();
           break;
 
         case DRUPAL_BOOTSTRAP_FULL:
+          if (function_exists('sb_stagemarker_message')) sb_stagemarker_message('BEFORE DRUPAL_BOOTSTRAP_FULL');
           require_once DRUPAL_ROOT . '/includes/common.inc';
           _drupal_bootstrap_full();
+          if (function_exists('sb_stagemarker_message')) sb_stagemarker_message('AFTER  DRUPAL_BOOTSTRAP_FULL');
           break;
       }
     }
